plugins/woocommerce-blocks/assets/js/blocks-registry/payment-methods/registry.ts

```diff
@@ -11,15 +11,20 @@ import type {
 	ExpressPaymentMethods,
 } from '@woocommerce/types';
 import { dispatch } from '@wordpress/data';
+import {
+	ActionCreatorsOf,
+	ConfigOf,
+	DispatchReturn,
+} from '@wordpress/data/build-types/types';
 
 /**
  * Internal dependencies
  */
 import { default as PaymentMethodConfig } from './payment-method-config';
 import { default as ExpressPaymentMethodConfig } from './express-payment-method-config';
 import { canMakePaymentExtensionsCallbacks } from './extensions-config';
-
-import { STORE_KEY as PAYMENT_STORE_KEY } from '../../data/payment/constants'; // Full path here because otherwise there's a circular dependency.
+import { STORE_KEY as PAYMENT_STORE_KEY } from '../../data/payment/constants';
+import type { PaymentStoreDescriptor } from '../../data/payment';
```

Comment:
We can't get `paymentStore` from `@woocommerce/block-data` here as it introduces a circular dependency. The registry is really tightly coupled with the payment data store, so untangling these to remove the circular dependency would be too much work. Instead, I am importing only the _type_ of the `paymentStore` and casting the type returned by `dispatch`. It's the same result, just a little messier. This is the only file I had to do this in.